custom_components/tahoma/__init__.py

```diff
@@ -76,13 +76,25 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry):
     username = entry.data.get(CONF_USERNAME)
     password = entry.data.get(CONF_PASSWORD)
 
+    client = TahomaClient(username, password)
+
     try:
-        client = TahomaClient(username, password)
         await client.login()
+    except TooManyRequestsException:
+        _LOGGER.error("too_many_requests")
+        await client.close()
```

Comment:
We forgot to close the client on exceptions.. I also added logging, since this try/catch will be called on a restart for example.